src/electionguard/chaum_pedersen.cpp

```diff
@@ -576,26 +576,34 @@ namespace electionguard
     unique_ptr<ConstantChaumPedersenProof>
     ConstantChaumPedersenProof::make(const ElGamalCiphertext &message, const ElementModQ &r,
                                      const ElementModP &k, const ElementModQ &seed,
-                                     const ElementModQ &hash_header, uint64_t constant)
+                                     const ElementModQ &hash_header, uint64_t constant,
+                                     bool shouldUsePrecomputedValues /* = false */)
     {
         Log::trace("ConstantChaumPedersenProof:: making proof");
         auto *alpha = message.getPad();
         auto *beta = message.getData();
 
-        // Derive nonce from seed and the constant string below
-        auto nonces = make_unique<Nonces>(seed, "constant-chaum-pedersen-proof");
         unique_ptr<ElementModQ> u;
 
         // Compute the NIZKP
         unique_ptr<ElementModP> a; //𝑔^𝑢 mod 𝑝
         unique_ptr<ElementModP> b; // 𝐾^𝑢 mod 𝑝
-        // check if the are precompute values rather than doing the exponentiations here
-        unique_ptr<Triple> triple = PrecomputeBufferContext::getTriple();
-        if (triple != nullptr) {
-            u = triple->get_exp();
-            a = triple->get_g_to_exp();
-            b = triple->get_pubkey_to_exp();
-        } else {
+
+        if (shouldUsePrecomputedValues) {
+            Log::debug("ConstantChaumPedersenProof:: using precomputed values. Your seed value is "
```

Comment:
this kept throwing me for a loop when iwas testing becauise i forgot that when we use the precompute buffers that the nonces are no longer deterministic. the API's around this behavior are pretty inconsistent and not very descriptive. For instance in several places we expect a `seed` or a `proofSeed` or a `nonceSeed` to be passed in as a required parameter, but it is not actually used when the precompute buffers are used. I made #216 to capture some of these concerns.